### PR TITLE
Add support for all types + refactor/cleanup

### PR DESCRIPTION
**Pull request by nezihyigitbasi**

### CODE REVIEW

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetVectorizedRecordReader.java [hunk 1]
```diff
@@ -1,251 +0,0 @@
...
-/**
- * Reads the records from a block of a Parquet file
- *
- * @see ParquetInputFormat
- *
- * @author Zhenxiao Luo
- *
- * @param <T> type of the materialized records
- */
-public class ParquetVectorizedRecordReader<T> extends RecordReader<Void, T>
```
Comment (dongche) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetVectorizedRecordReader.java]: Can we remove this reader? Or it is planned to add `nextBatch()` method in `ParquetRecordReader`? 
MR jobs may need this reader to read batch. Currently we have `ParquetReader` to read batches, but `ParquetInputFormat` does not provide a similar reader for MR based engine like Hive to use.

I could open an issue to impl it, if it make sense. What do you think? @nezihyigitbasi

Comment (nezihyigitbasi) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetVectorizedRecordReader.java]: @dongche we remove `ParquetVectorizedRecordReader` with this PR. And yep it makes sense to also support batch reads from `ParquetRecordReader` too. Currently both  `ParquetReader` and `ParquetRecordReader` uses the `InternalParquetRecordReader` class to do the reads and `InternalParquetRecordReader` already has a `nextBatch()` method, which you can use at `ParquetRecordReader`. Please go ahead if you want to implement it.

**Comment by nezihyigitbasi** [comment]
@zhenxiao updated the PR to add support for more primitive types

**Comment by nezihyigitbasi** [comment]
@zhenxiao @dongche this is a major rewrite of what we have right now. 
@dongche this also addresses your concern of returning more than 1K rows, now every call except for the row group boundaries should return 1K rows. Please take a look and any comments are welcome as usual.

**Comment by dongche** [comment]
Great! I will take a look, thanks for the quick updates.

**Comment by nezihyigitbasi** [comment]
@zhenxiao has merged this PR, so closing.

**Comment by nezihyigitbasi** [comment]
@dongche this is now merged to the `vector` branch so please go ahead and update your hive poc with this new API and let us know your feedback. thanks!

**Comment by dongche** [comment]
@nezihyigitbasi Great! I like this change!

I am working on it and will keep you posted.

**Pull request closed without merging**